internal/db/merge.go

```diff
@@ -58,24 +60,28 @@ func (db *db) executeMerge(ctx context.Context, dagMerge event.Merge) error {
 		return err
 	}
 
-	mt, err := getHeadsAsMergeTarget(ctx, txn, dsKey)
+	mt, err := getHeadsAsMergeTarget(ctx, txn, dsKey.WithFieldId(core.COMPOSITE_NAMESPACE))
```

Comment:
nitpick: I see you updated some `fieldId` to `fieldID`.

Missed this one from `func (k DataStoreKey) WithFieldId(fieldId string) DataStoreKey {`